freqtrade/optimize/backtesting.py

```diff
@@ -349,6 +350,11 @@ def backtest(self, processed: Dict,
         trades: List[LocalTrade] = []
         self.prepare_backtest(enable_protections)
 
+        # Update dataprovider cache
+        for pair, dataframe in processed.items():
+            self.dataprovider._set_cached_df(pair, self.timeframe, dataframe)
+        self.strategy.dp = self.dataprovider
```

Comment:
I am not sure why/how `strategy.dp` gets unset, but it does in hyperopt. This workaround does the job.. But there may be a way to fix this more proper.

Reply:
it's intentional as it contains an exchange object - which you can't submit to another process through pickle (pickle will complain and fail).

Reply:
in theory i guess having a "no exchange dataprovider" would do ... 
but it'll require some more changes to dataprovider (it should fail if one of the "requires exchange" functions is called ...).